crates/lsp/src/lsp.rs

```diff
@@ -814,8 +829,33 @@ impl LanguageServer {
         configuration: Arc<DidChangeConfigurationParams>,
         cx: &App,
     ) -> Task<Result<Arc<Self>>> {
+        /// This is a drop-in replacement for `lsp_types::Initialize`, that allows us to insert non-standard objects into `Params`.
+        struct InitializeExtended;
```

Comment:
Frankly, wrecking the entire type system for just clangd looks very unmotivating.

Why this time, they are were not able to use `experimental` field in the client capabilities?

https://github.com/zed-industries/zed/blob/878b50c991042b876f1fb67f5c76619cbca2170c/crates/languages/src/c.rs#L266-L279

Reply:
I originally tried advertising the capability via ‘experimental’, as you’re suggesting but clangd just ignored it :(

Reply:
I would propose to do something like https://github.com/llvm/llvm-project/pull/114699 in clangd first (if not done already), and then use the `capabilities.experimental` part in Zed for this.

Reply:
I was just looking into doing this, it seems such a change already exists for 4 months, but there has not yet been a stable release since.